example-app/SantokuApp/src/features/demo-push-notification/pages/PushNotificationSenderPage.tsx

```diff
@@ -0,0 +1,341 @@
+import {useGetFcmToken} from '../services/useGetFcmToken';
+import {useHasPermission} from '../services/useHasPermission';
+import {usePushNotificationSendCommands} from '../services/usePushNotificationSendCommands';
+import {usePushNotificationSelectPicker} from '../use-cases/usePushNotificationSelectPicker';
+
+const handleApiError = (e: unknown) => {
+  if (axios.isAxiosError(e)) {
+    const axiosError = e as AxiosError<ErrorResponse>;
+    if (axiosError.response) {
+      alert(axiosError.response.data.message);
+      return;
+    }
+    alert(e);
+  }
+};
+
+export type PushNotificationSenderPageProps = {
+  navigateToPushNotificationStatus: () => void;
+};
+export const PushNotificationSenderPage: React.FC<PushNotificationSenderPageProps> = ({
+  navigateToPushNotificationStatus,
+}) => {
+  const {
+    form,
+    setFormContentAvailable,
+    setFormPriority,
+    setFormInterruptionLevel,
+    setFormChannel,
+    addDataField,
+    removeDataField,
+  } = usePushNotificationSenderForm();
+  const {
+    priorities,
+    onSelectedPriorityChange,
+    interruptionLevels,
+    onSelectedInterruptionLevelsKeyChange,
+    channels,
+    onSelectedChannelChange,
+  } = usePushNotificationSelectPicker({setFormPriority, setFormInterruptionLevel, setFormChannel});
+  const {fcmToken, error: fcmTokenError} = useGetFcmToken();
+  const {permission, error: permissionError} = useHasPermission();
+  const {
+    sendToMe: callSendToMe,
+    isSendingToMe,
+    sendToAll: callSendToAll,
+    isSendingToAll,
+  } = usePushNotificationSendCommands();
+
+  if (fcmTokenError) {
+    log.trace(`Failed to get token. cause=[${String(fcmTokenError)}]`);
+  }
+  if (permissionError) {
+    log.trace(`Failed to get permission status. cause=[${String(permissionError)}]`);
+  }
+
+  const {data: accountData, isFetching: isFetchingAccountData} = useAccountData();
+
+  const isAllowedPermission = useMemo(
+    () =>
+      permission === messaging.AuthorizationStatus.AUTHORIZED ||
+      permission === messaging.AuthorizationStatus.PROVISIONAL,
+
+    [permission],
+  );
+
+  const hasFcmToken = useMemo(() => {
+    return !!fcmToken;
+  }, [fcmToken]);
+
+  const isRegisteredDeviceToken = useMemo(() => {
+    return fcmToken ? accountData?.account.deviceTokens.includes(fcmToken) : false;
+  }, [accountData?.account.deviceTokens, fcmToken]);
+
+  const isReceivableOnThisDevice = useMemo(
+    () => isAllowedPermission && hasFcmToken && isRegisteredDeviceToken,
+    [hasFcmToken, isAllowedPermission, isRegisteredDeviceToken],
+  );
+
+  const buildPushNotificationParams = useCallback(() => {
+    // string型の入力項目は、空文字の場合は送信しないため除外する
+    const filteredStringField = Object.entries(form.values)
+      .filter(([_, value]) => {
+        return typeof value === 'string' ? value : true;
+      })
+      .reduce((acc, [key, value]) => {
+        acc[key] = value;
+        return acc;
+      }, {} as {[key: string]: any});
```

Comment:
refactor: Object.fromEntries を使用
```suggestion
    const filteredStringField = Object.fromEntries(
      Object.entries(form.values).filter(([_, value]) => (typeof value === 'string' ? value : true)),
    );
```